command/rm.go

```diff
@@ -174,5 +190,17 @@ func validateRMCommand(c *cli.Context) error {
 		return fmt.Errorf("expected at least 1 object to remove")
 	}
 
-	return sourcesHaveSameType(c.Args().Slice()...)
+	src := c.Args().Slice()
+	if err := sourcesHaveSameType(src...); err != nil {
```

Comment:
Let's change the signature of this function and make it the same with `hasSameBuckets`
```
func hasSameTypes(urls []*url.URL) error 
```

No need to create storage.URL twice ([ref](https://github.com/peak/s5cmd/blob/333370fa083048a424066f1acffc8f01a409e484/command/rm.go#L165))

Reply:
with `func hasSameTypes(urls []*url.URL) error` it would be returning error whereas `hasSameBuckets` function returns a boolean (we might have slight inconsistency in naming pattern). maybe we can make it `func ensureSameTypes(urls []*url.URL) error`, is that ok?

Reply:
Sorry. Return value should be boolean rather than error. How about:

```
func hasSameTypes(urls []*url.URL) bool
```

Reply:
hmm, since `sourcesHaveSameType` returns more than one type of error messages, I am not sure how we can take those outside the function.

Reply:
Hmm, I thought it only validates source types (by considering the method name).

I think we should refactor and improve this validation logic:

1) In the current design, we need to iterate over all source URLs. I think it is better to validate URLs in one phase.

2) Method name of sourcesHaveSameType is wrong. It also validates whether the source URL is prefix or bucket.  

3) [Validation error here](https://github.com/peak/s5cmd/blob/333370fa083048a424066f1acffc8f01a409e484/command/rm.go#L172) is misleading. It raises an error if sources are bucket or prefix. However, the error message is "source argument must contain wildcard character", which is actually wrong.

Reply:
Does it make sense to you?

Reply:
Oh yeah 💯, that makes sense. thanks a lot.

Reply:
Thanks for your effort. This implementation is much better 👍

Reply:
thank you for your time and help :)